### ISSUE TITLE
Unable to mount folders with Parallels 14

### ISSUE THREAD

**Issue by dazwin** [Parallels/vagrant-parallels#325]
Here's a sample Vagrantfile:

```
# -*- mode: ruby -*-

VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box = "bento/ubuntu-16.04"

  config.vm.provider :parallels do |p, override|
    p.update_guest_tools = true
  end
end
```

Works fine on Parallels 13, but errors with the following on 14:
```
Bringing machine 'default' up with 'parallels' provider...
==> default: Registering VM image from the base box 'bento/ubuntu-16.04'...
==> default: Cloning new virtual machine...
==> default: Unregistering the box VM image...
==> default: Setting the default configuration for VM...
==> default: Checking if box 'bento/ubuntu-16.04' is up to date...
==> default: Setting the name of the VM: hidden_default_1535652911346_8247
==> default: Preparing network interfaces based on configuration...
    default: Adapter 0: shared
==> default: Clearing any previously set network interfaces...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 10.211.55.102:22
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection refused. Retrying...
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default: 
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for Parallels Tools installed on the VM...
==> default: Parallels Tools installed on this VM are outdated! In most cases
==> default: this is fine but in rare cases it can cause things such as shared
==> default: folders to not work properly. If you see shared folder errors,
==> default: please update Parallels Tools within the virtual machine and
==> default: reload your VM.
==> default: Installing the proper version of Parallels Tools. This may take a few minutes...
==> default: Parallels Tools have been installed. Rebooting the VM...
==> default: Attempting graceful shutdown of VM...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 10.211.55.102:22
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection refused. Retrying...
==> default: Machine booted and ready!
==> default: Mounting shared folders...
    default: /vagrant => ### hidden ###
Failed to mount folders in Linux guest. This is usually because
the "prl_fs" file system is not available. Please verify that
Parallels Tools are properly installed in the guest and
can work properly. If so, the VM reboot can solve a problem.
The command attempted was:

mount -t prl_fs -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t prl_fs -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant
```

If I remove the guest tools update, the VM boots successfully.

**Comment by legal90** [comment]
I can confirm this issue on Parallels Tools for Linux v14

It seems that `prl_fs` module in the newer version of guest tools doesn't support mounting the single device (shared folder) to the multiple targets.
By default Parallels Tools mounts all shared folders to `/media/psf/<folder_name>` and now vagrant fails to mount it to the desired point:

```
# mount | grep vagrant
vagrant on /media/psf/vagrant type prl_fs (rw,nosuid,nodev,noatime,sync)

# mount -t prl_fs -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant
mount: mount vagrant on /vagrant failed: File exists
```
@romankulikov do you know, is it an expected behavior? Also, it seems that `--shf-guest-automount` option doesn't work: Parallels Tools just ignore it and still mount all folders to `/media/psf/<folder_name>` when this option is set to `off`. Could it be fixed in the closest release?

Actually, our provider disables the Parallels Tools auto-mounting feature by [flushing `/usr/bin/prlfsmountd` script]
(https://github.com/Parallels/vagrant-parallels/blob/e4731621382e9bacb80dd073eada508fb69b0e8d/lib/vagrant-parallels/guest_cap/linux/mount_parallels_shared_folder.rb#L103-L109), but that hook is applied after the initial VM boot, so Parallels Tools manage to auto-mount all folders on the first time. That's actually what we have in this issue.

@dazwin I'll try to come up with a solution soon. As a temporary workaround I can suggest the following:
1. Just ignore the error and do `vagrant reload`. Vagrant shared folder mounting should work fine on all subsequent boots because our provider flushes `/usr/bin/prlfsmountd`, so Parallels Tools don't don't mess around.

OR

2. Remove or comment-out the line `p.update_guest_tools = true`, so Parallels Tools will not be upgraded. The previous version of Parallels Tools (< 14) should work just fine.

**Comment by nachopants** [comment]
Can confirm this is broken as well.

Bringing machine 'default' up with 'parallels' provider...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 0: shared
==> default: Clearing any previously set network interfaces...
==> default: Forwarding ports...
    default: 80 => 8080
    default: 443 => 8443
    default: 3306 => 3306
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 10.211.55.15:22
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Machine booted and ready!
==> default: Checking for Parallels Tools installed on the VM...
==> default: Parallels Tools were not detected on this VM! They are required
==> default: for shared folders, time sync and more. If shared folders are not
==> default: working on this machine, please install Parallels Tools within the
==> default: virtual machine and reload your VM.
==> default: Installing the proper version of Parallels Tools. This may take a few minutes...
==> default: Parallels Tools have been installed. Rebooting the VM...
==> default: Attempting graceful shutdown of VM...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 10.211.55.15:22
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Machine booted and ready!
==> default: Mounting shared folders...
    default: /vagrant => ##hidden##
Failed to mount folders in Linux guest. This is usually because
the "prl_fs" file system is not available. Please verify that
Parallels Tools are properly installed in the guest and
can work properly. If so, the VM reboot can solve a problem.
The command attempted was:

mount -t prl_fs -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t prl_fs -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

**Comment by legal90** [comment]
@romankulikov Thanks for your reply! Do you have any suggestions how can we work it around and prevent shared folders from auto-mounting to `/media/pfs/` ? 
Or we should wait for the fix on Parallels Tools side?

**Comment by romankulikov** [comment]
@legal90 well, trivial workaround is to get rid of `/usr/bin/prlfsmountd` (rename it or replace to somewhere) in the guest. But, of course, better is to wait for a fix on a Parallels Desktop side.

**Comment by legal90** [comment]
> trivial workaround is to get rid of /usr/bin/prlfsmountd
@romankulikov Yeah, that's what we already do. The thing is that Parallels Tools daemon starts pretty early, so it executes this script on the initial boot _before_ we modify it. 
Okay, we'll wait for the fix on the Parallels Desktop side.

Also, I think we should send a patch to Packer (to its `parallels` builder), so `/usr/bin/prlfsmountd` will be disabled on the box build step.

**Comment by getsource** [comment]
Just a quick note that I was bit by this as well and spent a long time figuring it out before rolling back to Parallels 13. Thanks folks in here for pointing out the bug!

**Comment by max-arnold** [comment]
Have the same issue, but the second boot is fine (as in #133). The machine config is minimal:

```
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu-18.04"
end
```
The image I use was prepared with `packer build -only=parallels-iso ubuntu-18.04-amd64.json` using Bento boxes: https://github.com/chef/bento/tree/master/ubuntu.

Parallels 14.0.1, Vagrant  2.1.5, vagrant-parallels 1.7.8.

**Comment by max-arnold** [comment]
@legal90 I've submitted a patch to bento, feel free to review: https://github.com/chef/bento/pull/1122

**Comment by legal90** [comment]
@max-arnold Thank you! Yes, it should fix the issue in `bento/*` boxes.

**Comment by nachopants** [comment]
Anyone know if Parallels desktop is fixed for this yet?

**Comment by romankulikov** [comment]
I hope the fix is coming with one of the nearest updates.

**Comment by romankulikov** [comment]
> > @romankulikov do you know, is it an expected behavior?
> 
> Confirm the problem with mounting one shared folder twice. This is definitely a bug appeared in PD14.

This is fixed in 14.1.0 update.
